lang/LangPrimSource/SC_Ableton_Link.cpp

```diff
@@ -0,0 +1,283 @@
+#ifdef SC_ABLETON_LINK
+
+#include "PyrKernel.h"
+#include "PyrSched.h"
+#include "GC.h"
+#include "PyrPrimitive.h"
+#include "PyrSymbol.h"
+
+#include "SCBase.h"
+#include "SC_Clock.hpp"
+
+#include <ableton/Link.hpp>
+
+static std::chrono::microseconds linkTimeOfInitialization;
+void initLink()
+{
+	linkTimeOfInitialization   = ableton::link::platform::Clock().micros();
+}
+
+inline std::chrono::microseconds hrToLinkTime(double secs){
+	auto time = std::chrono::duration<double>(secs);
+	return std::chrono::duration_cast<std::chrono::microseconds>(time) + linkTimeOfInitialization;
+}
+
+inline double linkToHrTime(std::chrono::microseconds micros){
+	return DurToFloat(micros - linkTimeOfInitialization);
+}
+
+class LinkClock : public TempoClock
```

Comment:
a line or two of documentation here couldn't hurt

Reply:
I'm afraid I'm not familiar with C++ inline documentation practices. I have no idea what you want here. Would you mind taking this one?

Reply:
I can comment through a description of the functionality - in case someone is not familiar with the Ableton Link API ?  Not sure if that's what you had in mind though...